### ISSUE TITLE
Missing aria-label in nav

### ISSUE THREAD

**Issue by Alena5041** [facebook/docusaurus#8308]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

In any docusaurus application including the official docusaurus website ([https://docusaurus.io/docs](https://docusaurus.io/docs)), there are two `<nav>` elements, that don’t have `aria-label`.

### Steps to reproduce

1. Go to [https://docusaurus.io/docs](https://docusaurus.io/docs)
2. Open dev tools
3. Search for `<nav>` elements (there are 4 `<nav>` elements in total)
4. First and second `<nav>` elements don’t have `aria-label`

### Expected behavior

All `<nav>` elements have `aria-label` so that assistive technologies like screen readers can identify unique navigation elements.

### Actual behavior

Main navigation and sidebar navigation elements don’t have an `aria-label`. 
Currently, this is causing accessibility issues: [https://dequeuniversity.com/rules/axe/4.4/landmark-unique](https://dequeuniversity.com/rules/axe/4.4/landmark-unique).

### Your environment

- Docusaurus version used: 2.2.0

### Self-service

- [X] I'd be willing to fix this bug myself.

**Comment by slorber** [comment]
Thanks for reporting a11y issues.

I'm not an a11y expert, what do you suggest as aria-label here?

Is there a way to detect such issues easily as part of our CI?

**Comment by Alena5041** [comment]
You're welcome. 

For main navigation, it can be `aria-label="Main"`, for sidebar `aria-label="Sidebar"`.

You can use [Axe](https://github.com/dequelabs/axe-core) to detect such issues, it is an accessibility testing engine for websites and other HTML-based user interfaces.

**Comment by Alena5041** [comment]
It might be better if it would be possible to set the `aria-label` by a prop, the same way how you set it for navbar items in the docusaurus.config.js, just to give users a bit more control over the text content. Would you be interested in implementing it yourself or should I do it?

**Comment by slorber** [comment]
Hi,

All aria labels should be translated and we provide default translations for them. If label is provided through config then it makes it harder for users to localized those, and I doubt many users will want to customize this anyway so it should probably not be an option.

The way to customize it would be:
- to swizzle the component
- to use the translation strings (you can customize the default english translations for example)

**Comment by Alena5041** [comment]
Hi, 

Thank you for the translation tip, I didn’t know about it! I agree with you, then we can just add the `aria-label` for the main and sidebar navbars, and it would be possible to change the values by translation strings.

**Comment by Alena5041** [comment]
Hi,

Would you be interested in implementing it yourself or should I do it?

**Comment by slorber** [comment]
Hi

It's an easy to contribute issue so I keep it for anyone willing to send a PR ;)

**Comment by r-lawrence** [comment]
Hi @slorber and @Alena5041.  Im a first time contributor who navigated to this bug and decided to give it a shot.  I have created a draft [PR](https://github.com/facebook/docusaurus/pull/8366) for the issue and had some questions regarding if I processed this correctly.

when looking into this issue, I realized in the comments that "aria labels should be translated".  Im not entirely sure I understand what translated means, but when looking through the repo other `aria-label` attributes are made using the `translate` function from `@docusaurus/Translate`.  

Each instance of a translate function accepted an `Id` Parameter.  I tried to follow the naming convention similarities I noticed, but am unsure if they have been named appropriately.

I have created a draft PR but am unsure if I completed the fix appropriately

- based on the work I did in the PR, did I attach the correct `Id` parameter?

- I noticed tests for the translate function were established in the `packages/docusaurus/src/client/exports/__tests__/Translate.test.tsx` file so I didnt know if any additional testing would be required.  Should I have written tests somewhere to verify the `nav` element is rendered with an `aria-label`? 

Thanks a bunch in advance for the response/feedback.

**Comment by slorber** [comment]
Thanks @r-lawrence 👍 

@Alena5041 can you please help review the PR, added some questions/concerns about those aria labels.

Also wondering if we do the right thing on mobile? We don't use `nav` here for the docs mobile sidebar apparently so there's no reported missing aria-label. Do screen reader users also use Docusaurus with a mobile viewport?

**Comment by Alena5041** [comment]
Hi @slorber,

Sure, screen reader users also use a mobile viewport. The fact that you change for mobile `<nav>`  for the `<div class="navbar-sidebar">` is not a problem in your case, because the main and sidebar navigations wrap into one main `<nav>`. But still, need to have `aria-label` on the main navigation.

**Comment by Alena5041** [comment]
Thanks @slorber and @r-lawrence for the quick solution to my issue
